src/app/chart/donut/donut.component.ts

```diff
@@ -0,0 +1,144 @@
+import { Component, DoCheck, Input, OnInit } from '@angular/core';
+
+import { cloneDeep, defaults, isEqual, merge } from 'lodash';
+import { ChartDefaults } from '../chart.defaults';
+import { ChartBase } from '../chart.base';
+import { DonutConfig } from './donut-config';
+
+import * as d3 from 'd3';
+
+@Component({
+  selector: 'pfng-chart-donut',
+  templateUrl: './donut.component.html'
+})
+export class DonutComponent extends ChartBase implements DoCheck, OnInit {
+
+  @Input() chartData: any;
+  @Input() config: DonutConfig;
+
+  public donutChartId: any;
+
+  private prevChartData: any;
+  private defaultConfig: DonutConfig;
+
+  /**
+   * Default constructor
+   * @param chartDefaults
+   */
+  constructor(private chartDefaults: ChartDefaults) {
+    super();
+  }
+
+  ngOnInit(): void {
+    this.donutChartId = 'donutChart';
+    if (this.config.chartId) {
+      this.donutChartId = this.config.chartId + this.donutChartId;
+    }
+
+    this.chartLoaded.subscribe({
+      next: (chart: any) => {
+        this.chartAvailable(chart);
+      }
+    });
```

Comment:
If we subscribe, we also need to unsubscribe when the component is destroyed. Otherwise, it will ultimately contribute to memory leaks as we discovered in OSIO.

In OSIO, we created an array, named subscriptions. The code looks something like this:

subscriptions: Subscription[] = [];
...
constructor() {
    this.subscriptions.push(this.chartLoaded.subscribe...
...

 ngOnDestroy(): void {
    this.subscriptions.forEach(sub => {
      sub.unsubscribe();
    });
  }

Reply:
Nice catch. Done.